.github/workflows/create-release.yml

```
@@ -0,0 +1,34 @@
+name: Create Release (Versioned)
+
+on:
+  workflow_run:
+      workflows: ["Publish NPM Package"]
+      types:
+        - completed
+
+jobs:
+  create-release:
+    # only run if the "Publish NPM Package" workflow was successful
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Setup Node
+      uses: actions/setup-node@v3
+    - name: Git Fetch       # Necessary for the changelog to generate everything as intended
+      run: git fetch --prune --unshallow --tags
+    - name: Set Current Package Version as a Variable
+      id: package-version   # Can be referenced with ${{ steps.package-version.outputs.current-version }}
+      uses: martinbeentjes/npm-get-version-action@main
+    - name: Create Changelog Config
+      # For more info on the auto-changelog config params: https://www.npmjs.com/package/auto-changelog/v/2.4.0
+      run: echo '{"output":"CHANGELOG.md","template":"keepachangelog","package":true,"unreleased":false,"commitLimit":false,"sortCommits":"date-desc"}' > .auto-changelog
+    - name: Generate Changelog
+      run: npx auto-changelog@2.4.0
+    - name: Create Release
+      uses: ncipollo/release-action@v1
+      with:
+        # This CHANGELOG.md file will generated by the "auto-changelog" NPM package. This setting assumes that there is a ".auto-changelog" file in the root of the consuming repository with a key value pair of: "output": "CHANGELOG.md".
```

Updating comment for clarity
```suggestion
        # This CHANGELOG.md file will generated by the "auto-changelog" NPM package. This setting assumes that there is a ".auto-changelog" file in the root of the consuming repository with a key value pair of: "output": "CHANGELOG.md", which is generated in the above "Create Changelog Config" step.
```